assembl/fabfile.py

```diff
@@ -2898,17 +2902,14 @@ def set_ftp_private_information(force=False):
     execute(install_ncftp_client)
     if not exists('ncftp.cfg') or force:
         # fill template and set file permission
-        fill_template('assembl/templates/system/ncftp.cfg.jinja2', env, 'ncftp.cfg')
+        with NamedTemporaryFile(delete=False) as f:
+            ftp_info_file = f.name
+        fill_template('assembl/templates/system/ncftp.cfg.jinja2', env, ftp_info_file)
         path = join(env.projectpath, 'ncftp.cfg')
         try:
-            put_status = put('ncftp.cfg', path)
-            if put_status.failed:
-                raise RuntimeError('The put operation failed to put ncftp.cfg')
-
+            put(ftp_info_file, path)
             # Only readable by the user
             run('chmod 400 %s' % path)
             run('chown %(user)s:%(user)s %(file)s' % {'user': env.user, 'file': path})
-
-        finally:
-            # Remove the templated file
-            os.unlink('ncftp.cfg')
+        except:
+            raise Exception('The put operation failed to put ncftp.cfg')
```

Comment:
You still need to unlink the file, because we create NamedTemporaryFile with delete=False